plugins/compile.sh

```diff
@@ -7,18 +7,268 @@
 #
 # This file is part of AMX Mod X.
 
-# new code contributed by \malex\
-
 test -e compiled || mkdir compiled
 rm -f temp.txt
 
-for sourcefile in *.sma
-do
-        amxxfile="`echo $sourcefile | sed -e 's/\.sma$/.amxx/'`"
-        echo -n "Compiling $sourcefile ..."
-        ./amxxpc $sourcefile -ocompiled/$amxxfile >> temp.txt
-        echo "done"
-done
+os="$(uname -s)"
+case "$os" in
+	"Linux"  ) os="linux" ;;
+	"Darwin" ) os="mac" ;;
+	*)
+esac
+
+quit_script(){
+	echo "Exiting..."
+	exit 1
+}
+
+check_if_gameserver () {
+	if [ ! -d "../configs" -a ! -d "../data" -a ! -d "../dlls" -a ! -d "../modules" -a ! -d "../plugins" ]
+	then
+		echo "Make sure that folders configs, data, dlls, modules and plugins"
+		echo "  exist and run the script again"
+		exit 1
+	fi
+}
+
+get_base () {
+	tar=(which tar)
+	if (( tar == -1 ))
+	then
+		echo "Please install \"tar\" and run the script again"
+		
+		if [ "$os" == "linux" ]
+		then
+			echo "  type: sudo apt-get install tar"
```

Comment:
At least one apt based distribution lacks tar?
Having tar is standard for apt and is in depends.